odoo/tools/query.py

```diff
@@ -69,6 +69,7 @@ def __init__(self, env, alias: str, table: (SQL | None) = None):
 
         # groupby, having, order, limit, offset
         self.groupby: SQL | None = None
+        self._order_groupby: list[SQL] = []
```

Comment:
`_order_groupby` seems like a temporary data which should be handled outside of `Query` object. Maybe addd an `assert not self._order_groupby` in the `Query.select` to make sure the attribute has been handled correctly?